cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,40 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
+  let user;
+
   before(() => {
-    
+    cy.visit('/');
+    cy.task('generateUser').then(generatedUser => {
+      user = generatedUser;
+    });
   });
 
-  it('', () => {
-    
+  it('should fill all fields',() => {
+    cy.findByPlaceholder('First Name').type(user.firstName);
+    cy.findByPlaceholder('Last Name').type(user.lastName);
+    cy.get('#userEmail').type(user.email);
+    cy.findByRadio('radio').check(user.gender, { force: true});
+    cy.findByPlaceholder('Mobile Number').type(user.phoneNumber);
```

Comment:
Instead of using `cy.findByRadio()`, which is not a part of the official Cypress command API, consider using `cy.get()` with a more specific attribute like `data-testid` or `id`. For example, you can use `cy.get('[data-testid=gender]')` if you add the `data-testid` attribute to the corresponding input field in the HTML.